### PR TITLE
Fix issues with correlation instrumentation

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

Issues from: https://github.com/PostHog/posthog/pull/6788#pullrequestreview-796785608

> The offset in the visibility sensor is not working. props.offset is never set. Couldn't figure out why this is the case, it's just not being passed to the logic correctly.

> We report correlation viewed (and related events) multiple times even if it's the same analysis. We should only report once and re-report only on refreshes/navigation, or if the funnel filters change.

> We only report correlation analyzed if the user does not scroll again at all in the 10s period, yet it's a pretty common case that you scroll a little bit while still "analyzing" the correlation analysis table. We should report the analyzed events if the table stays within the view for 10 seconds.

## How did you test this code?

Manually check events are fired

### CODE REVIEW

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 1]
```diff
@@ -371,6 +371,30 @@ export const funnelLogic = kea<funnelLogicType>({
                 },
             },
         ],
+        shouldReportCorrelationViewed: [
+            true as boolean,
+            {
+                loadResultsSuccess: () => true,
```
Comment (neilkakkar) [inline frontend/src/scenes/funnels/funnelLogic.ts:377]: This is an implicit action meaning filters have changed

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 2]
```diff
@@ -1198,24 +1222,22 @@ export const funnelLogic = kea<funnelLogicType>({
...
             breakpoint: BreakPointFunction
         ) => {
-            if (visible) {
+            if (visible && values.shouldReportPropertyCorrelationViewed) {
                 eventUsageLogic.actions.reportCorrelationViewed(values.filters, 0, true)
+                await breakpoint(10000)
+                eventUsageLogic.actions.reportCorrelationViewed(values.filters, 10, true)
             }
-
-            await breakpoint(10000)
```
Comment (neilkakkar) [inline frontend/src/scenes/funnels/funnelLogic.ts:1217]: the problem you mentioned with "analyzed" wasn't that it was happening 10 seconds after scroll, but that after the "not visible" event was fired, we were still firing the delayed breakpoint 🤦

File: frontend/src/scenes/insights/FunnelCorrelation.tsx [hunk 3]
```diff
@@ -52,7 +50,7 @@ export const FunnelCorrelation = (): JSX.Element | null => {
     }
 
     return (
-        <VisibilitySensor id={correlationPropKey} offset={150}>
+        <>
```
Comment (hazzadous) [inline frontend/src/scenes/insights/FunnelCorrelation.tsx:53]: non-blocking

guess we don't need any fragment wrapper ?

**Review by hazzadous — APPROVED** [review]
Not super clear what's going on, but looks good to go.

**Comment by neilkakkar** [comment]
@paolodamico I'll merge this, but will wait for you to test it out before removing `beta-` from events! Will start releasing right after.

**Pull request merged**